userspace/libsinsp/tracer_emitter.cpp

```diff
@@ -116,8 +116,10 @@ void tracer_writer::close_fd()
 	}
 }
 
-tracer_emitter::tracer_emitter(std::string tag)
+tracer_emitter::tracer_emitter(std::string tag, uint64_t timeout_ns)
 	: m_tag(std::move(tag))
+	, m_start_ns(sinsp_utils::get_current_time_ns())
```

Comment:
I think it would be good to allow providing an external timestamp for now. I remember in the past where lots of calls to sinsp_utils::get_current_time_ns() ended up being expensive.